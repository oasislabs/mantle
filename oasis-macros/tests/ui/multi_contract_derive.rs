#![feature(proc_macro_hygiene)]
#[oasis_std::contract]
mod contract {
    #[derive(Contract)]
    pub struct Counter(u32);

    #[derive(Contract)]
    pub struct Counter2(u32);

    impl Counter {
        pub fn new(ctx: &Context) -> Result<Self> {
            Ok(Self(42))
        }
    }

    impl Counter2 {
        pub fn new(ctx: &Context) -> Result<Self> {
            Ok(Self(42))
        }
    }
}

fn main() {}
