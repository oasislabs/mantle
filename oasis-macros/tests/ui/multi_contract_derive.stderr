error: `contract!` must contain exactly one #[derive(Contract)] struct. Additional occurrence here:
 --> $DIR/multi_contract_derive.rs:7:5
  |
7 |     #[derive(Contract)]
  |     ^

error: aborting due to previous error

